erizo/src/erizo/WebRtcConnection.cpp

```diff
@@ -741,4 +741,20 @@ void WebRtcConnection::setTransport(std::shared_ptr<Transport> transport) {  //
   bundle_ = true;
 }
 
+std::pair<RTPExtensionsMap, RTPExtensionsMap> WebRtcConnection::getSourceExtensionMap() {
+  auto result = std::make_pair(extension_processor_.getVideoExtensionMap(),
+                               extension_processor_.getAudioExtensionMap());
+  return result;
+}
+
+void WebRtcConnection::setSourceExtensionMap(std::shared_ptr<WebRtcConnection> source_wrtc) {
+  std::weak_ptr<WebRtcConnection> this_weak = shared_from_this();
+  source_wrtc->asyncTask([this_weak](std::shared_ptr<WebRtcConnection> source_connection) {
+    auto map = source_connection->getSourceExtensionMap();
+    if (auto this_connection = this_weak.lock()) {
+      this_connection->extension_processor_.setSourceExtensionMap(map);
```

Comment:
I think we should call this_connection->asyncTask() here to make it run in the right worker.